src/support/err.c

```diff
@@ -198,7 +198,7 @@ __eventv_unpack_json_str(u_char *dest, size_t dest_len, char *src, size_t src_le
 static int
 __eventv_gen_msg(WT_SESSION_IMPL *session, char *buffer, size_t *buffer_len, bool is_json,
   int error, const char *func, int line, WT_VERBOSE_CATEGORY category, WT_VERBOSE_LEVEL level,
```

Comment:
This is something from the previous work in WT-8258, but I wonder why `size_t *buffer_len` is passed indirectly, rather than `size_t buffer_len`?  We generally use non-const pointers for return args, so it seems confusing.  It would allow the caller to be cleaner, generally you don't need a separate `remain` variable.

Reply:
Hi @ddanderson, that's a good point and I agree. I tried a first attempt in 4b8ffd4afca9cfe9b8553e43ebfc8b77192b36b7 which I think did what you are suggesting but I had to move that section of code inside `__eventv_gen_msg`:

```
    /*
     * The buffer is fixed sized, complain if we overflow. (The test is for no more bytes remaining
     * in the buffer, so technically we might have filled it exactly.) Be cautious changing this
     * code, it's a recursive call.
     */
    if (ret == 0 && remain == 0)
        __wt_err(
          session, ENOMEM, "error or message truncated: internal WiredTiger buffer too small");
```
And if this is executed, `__wt_err` will end up calling `__eventv_gen_msg` again and I don't think this is the best design. I reverted that change.

I tried a second approach in e0c64c48d9d2a9f2c7028899949b5bf69fddaea3 where instead of giving the `buffer_len` to `__eventv_gen_msg`, we are now giving a reference of the buffer remainder that is updated inside `__eventv_gen_msg`. Let me know what you think of this approach.

Reply:
@etienneptl, it looks like the current interface uses `size_t *remain`, but the question is the same.  Why is this argument by reference, and not just `size_t remain`?  It looks like the value pointed to by `remain` is never changed.  Am I missing something?

Reply:
Yes, `remain` is modified by every call to `WT_ERROR_APPEND` and there are a few in `__eventv_gen_msg`. The `remain` value is then tested later in `__eventv`:
```
    if (ret == 0 && remain == 0)
        __wt_err(
          session, ENOMEM, "error or message truncated: internal WiredTiger buffer too small");
```

Reply:
Ah, thank you.  Makes sense now.